### PR TITLE
Allow users for supplying their custom sane config

### PR DESCRIPTION
**Pull request by Erick555**
Fixes https://github.com/flathub/org.kde.skanlite/issues/14

### CODE REVIEW

File: org.kde.skanlite.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     ],
     "finish-args": [
         "--device=all",
+        "--env=SANE_CONFIG_DIR=/var/config/sane.d:",
```
Comment (travier) [inline org.kde.skanlite.json:16]: Wouldn't that mean that you need to be an admin to setup this config? I would expect unprivileged users to be able to setup the config on their own thus the need for the wrapper (https://github.com/flathub/org.kde.skanlite/issues/14#issuecomment-2310681109) as the path is dynamic (includes the home dir) and can not be set in the manifest here.

Comment (Erick555) [reply org.kde.skanlite.json:16]: Inside flatpak sandbox `/var/config` = `$XDG_CONFIG_HOME` (bind-mount). It's same path but static which is why we don't need a wrapper.

**Comment by mbugni** [comment]
Just tested using my scanner, and it works:
```shell
[📦 org.kde.skanlite ~]$ env | grep SANE
SANE_CONFIG_DIR=/var/config/sane.d:

[📦 org.kde.skanlite ~]$ cat /var/config/sane.d/net.conf 
# This is the net backend config file.

## saned hosts
# Each line names a host to attach to.
# If you list "localhost" then your backends can be accessed either
# directly or through the net backend.  Going through the net backend
# may be necessary to access devices that need special privileges.
localhost

[📦 org.kde.skanlite ~]$ scanimage -L
device `v4l:/dev/video0' is a Noname HP TrueVision HD Camera: HP Tru virtual device
device `snapscan:libusb:001:013' is a Acer FlatbedScanner22 flatbed scanner
device `net:localhost:snapscan:libusb:001:013' is a Acer FlatbedScanner22 flatbed scanner

[📦 org.kde.skanlite ~]$ rm /var/config/sane.d -rf
[📦 org.kde.skanlite ~]$ scanimage -L
device `v4l:/dev/video0' is a Noname HP TrueVision HD Camera: HP Tru virtual device
device `snapscan:libusb:001:013' is a Acer FlatbedScanner22 flatbed scanner
```

@travier you don't need any special privilege to edit file in your sandoboxed `config/sane.d` folder (like I did).

The only drawback without a wrapper is that I don't know how to create e a default empty `sane.d` folder. But since it is an advanced user config, I think it's ok to create it manually.

**Comment by mbugni** [comment]
@Erick555 @travier is it possible to add the following in the `.gitignore` file?
```
.flatpak-builder
```
It's just for local builds.

File: org.kde.skanlite.json (hunk 1 above)
Comment (travier) [reply org.kde.skanlite.json:16]: Awesome, I didn't know that. Thanks

**Comment by travier** [comment]
> is it possible to add the following in the `.gitignore` file?
> 
> ```
> .flatpak-builder
> ```
> 
> It's just for local builds.

Feel free to make another PR with that change.

**Comment by travier** [comment]
OK, looks great then. Let's add a some README to this repo with examples commands to set this up?

Something like:

```
mkdir -p $HOME/.var/app/org.kde.skanlite/config/sane.d
...
```

For some weird reason, https://github.com/flathub/org.kde.skanlite/issues/3#issuecomment-2298647550 is gone so I can't look up the commands/config anymore.

**Comment by mbugni** [comment]
@Erick555 I opened [this PR](https://github.com/Erick555/org.kde.skanlite/pull/1) to add a README and a `.gitignore`.

**Comment by Erick555** [comment]
@mbugni  plesase open this PR against flathub repo here, not mine.

**Review by travier — APPROVED** [review]

**Comment by travier** [comment]
Let's merge this and we'll merge the docs after.

@mbugni For the docs, no need to explain how to build the app (this is the same for all Flatpaks). Only how to configure the app for this.

**Pull request merged**